service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ProxyOptions.java

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.azure.sdk.iot.service;
+
+import lombok.Getter;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+
+/**
+ * The settings supported by this SDK when communicating to IoT Hub through a proxy. HTTP proxies are supported by this SDK,
+ * but only if the proxy does not require authentication.
+ */
+public class ProxyOptions
+{
+    @Getter
+    private Proxy proxy;
+
+    /**
+     * Create the proxy options for connecting to a proxy
+     * @param proxy the proxy to communicate through
+     */
+    public ProxyOptions(Proxy proxy)
```

Comment:
More of a question that PR comments: Does java support only setting proxies explicitly via the application, or does it also automatically tunnel traffic through a proxy set on the OS directly?

Reply:
What I mean is, if I have both set, who wins?

Reply:
Java doesn't allow you to force a proxy like that at the OS level as far as I know. Since we use transport libraries for HTTP and AMQP, we have to set the proxy at the http/amqp client level

Reply:
So if I understand correctly, if my OS would have some default proxy configured, that would never be honored?

Reply:
Yes, hence why customer's want proxy support on this SDK, instead of just setting it on their own devices

Reply:
Interesting!